prez/sparql/methods.py

```diff
@@ -1,223 +1,132 @@
+import asyncio
 import logging
+from typing import Dict, Tuple, Union
+from typing import List
 
-from typing import Dict, List, Tuple, Union
-
-from httpx import Client, HTTPError, HTTPStatusError
+import httpx
+from httpx import Client, AsyncClient
 from httpx import Response as httpxResponse
-from rdflib import Graph
-from starlette.responses import PlainTextResponse
+from rdflib import Namespace, Graph
+from starlette.requests import Request
 
 from prez.config import settings
-from prez.services.triplestore_client import sparql_clients
 
-log = logging.getLogger(__name__)
+PREZ = Namespace("https://prez.dev/")
 
-TIMEOUT = 30.0
 
+async_client = AsyncClient(
+    base_url=settings.sparql_endpoint,
+    auth=(settings.sparql_username, settings.sparql_password)
+    if settings.sparql_username
+    else None,
+)
 
-def sparql_query_non_async(query: str, prez: str) -> Tuple[bool, Union[List, Dict]]:
-    """Executes a SPARQL SELECT query for a single SPARQL endpoint"""
+client = Client(
+    base_url=settings.sparql_endpoint,
+    auth=(settings.sparql_username, settings.sparql_password)
+    if settings.sparql_username
+    else None,
+)
```

Comment:
Can we not use `base_url`? When relying on the client's `base_url` value as the actual URL, it adds a trailing slash. See the description of this issue in https://github.com/RDFLib/prez/pull/125.

This is the same issue as reported in https://github.com/RDFLib/prez/issues/95.

Reply:
resolved in commits below